### PR TITLE
refactored samples to use latest utility methods and improved those also

### PR DESCRIPTION
**Pull request by olensmar**

### CODE REVIEW

**Comment by olensmar** [comment]
@spindelmanne ping... :-)

File: modules/facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java [hunk 1]
```diff
@@ -44,15 +43,22 @@ public static RecipeExecutionResult executeRecipe(TestStepBuilder... testStepBui
...
-        if (executor == null) {
-            executor = RecipeExecutorBuilder.buildDefault();
+    /**
+     * Executes the specified TestRecipe and returns the result
+     *
+     * @param recipe the recipe to execute
+     * @return the excution result
+     */
+    public static RecipeExecutionResult executeRecipe(TestRecipe recipe) {
+        synchronized (Object.class) {
```
Comment (spindelmanne) [inline modules/facade/src/main/java/com/smartbear/readyapi4j/facade/execution/RecipeExecutionFacade.java]: I think we should synchronize on the RecipeExecutionFacade class instead.

File: modules/samples/java/src/test/java/com/smartbear/readyapi4j/samples/java/SimpleSoapTest.java [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
-        SoapRequestStepBuilder soapRequest = soapRequest(new URL("http://www.webservicex.com/globalweather.asmx?WSDL"))
+        RecipeExecutionResult result = executeRecipe(
+            soapRequest(new URL("http://www.webservicex.com/globalweather.asmx?WSDL"))
             .forBinding("GlobalWeatherSoap12")
             .forOperation("GetWeather")
             .withParameter("CountryName", "Sweden")
             .withPathParameter("//*:CityName", "Stockholm")
             .assertSoapOkResponse()
-            .assertSchemaCompliance();
+                .assertSchemaCompliance()
```
Comment (spindelmanne) [inline modules/samples/java/src/test/java/com/smartbear/readyapi4j/samples/java/SimpleSoapTest.java]: Intentional?

File: modules/core/src/main/java/com/smartbear/readyapi4j/support/AssertionUtils.java [hunk 3]
```diff
@@ -24,18 +25,33 @@ public static void assertExecution(Execution execution) {
...
+    }
+
+    public static void assertExecutionResult(RecipeExecutionResult execution) {
+        assertNotNull(execution);
+
+        if (LOG.isDebugEnabled()) {
+            logExecution(execution);
+        }
+
+        assertEquals(Arrays.toString(execution.getErrorMessages().toArray()),
```
Comment (spindelmanne) [inline modules/core/src/main/java/com/smartbear/readyapi4j/support/AssertionUtils.java]: Maybe prepend something like "Execution failed: " to the error message.

**Comment by olensmar** [comment]
thanks @spindelmanne - fixed!

**Pull request merged**